spec/alephant/logger/json_spec.rb

```diff
@@ -1,137 +1,92 @@
 require 'date'
 require 'spec_helper'
 require 'alephant/logger/json'
+require 'alephant/logger/level'
+require_relative 'support/json_shared_examples'
 
 describe Alephant::Logger::JSON do
-  subject { described_class.new(log_path) }
-
   let(:fn)       { -> { 'foo' } }
   let(:log_path) { '/log/path.log' }
-  let(:log_file) { instance_double File }
+  let(:log_output_obj) { instance_double(File) }
+  let(:msg) { :write }
 
   before do
-    allow(File).to receive(:open) { log_file }
-    allow(log_file).to receive :sync=
+    allow(File).to receive(:open) { log_output_obj }
+    allow(log_output_obj).to receive :sync=
   end
 
-  shared_examples 'JSON logging' do
-    let(:log_hash) do
-      { 'foo' => 'bar', 'baz' => 'quux' }
-    end
-
-    it 'writes JSON dump of hash to log with corresponding level key' do
-      allow(Time).to receive(:now).and_return('foobar')
+  logging_levels = Alephant::Logger::LEVELS.map(&:to_s)
+  levels_size = logging_levels.size - 1
```

Comment:
Slightly misleading here, as levels_size is the size minus one. I can see why you have done this to match up with the 0 based index.

It looks like you can do this, which can be from an index starting from 1:
```rb
%w(debug info warn error).to_enum.with_index(1).each do |level, i|
end
```

> http://ruby-doc.org/core-2.1.2/Enumerator.html

Then we can just use `logging_levels.size` later on, which is more accurate.

Reply:
Aaah, good shout. 👍